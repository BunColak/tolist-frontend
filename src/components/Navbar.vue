<template>
  <nav class="p-4 bg-indigo-700 text-white">
    <div class="mx-auto flex justify-between items-center">
      <div class="flex">
        <router-link to="/" class="font-semibold text-lg">ToList</router-link>
        <div>
          <router-link
            v-if="!this.$root.isLoggedIn"
            to="/features"
            class="text-lg ml-8"
            active-class="font-bold"
            >Features</router-link
          >
          <router-link
            to="/templates"
            class="text-lg ml-4"
            active-class="font-bold"
            >Templates</router-link
          >
        </div>
      </div>
      <div id="login-buttons" v-if="!this.$root.isLoggedIn">
        <button id="login" class="mx-4" @click="onLoginClick">Login</button>
        <button id="register" class="mx-4 border-2 px-2 py-1 rounded">
          Signup
        </button>
      </div>
    </div>
    <login v-show="showLogin && !this.$root.isLoggedIn" @close="onHideLogin" />
  </nav>
</template>

<script>
import Login from './Login'

export default {
  name: 'Navbar',
  components: {
    Login
  },
  data () {
    return {
      showLogin: false
    }
  },
  methods: {
    onLoginClick () {
      this.showLogin = true
    },
    onHideLogin () {
      this.showLogin = false
    }
  }
}
</script>
