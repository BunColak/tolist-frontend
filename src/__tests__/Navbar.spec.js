import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import Navbar from '@/components/Navbar'

describe('Navbar', () => {
  it('should render without problems', () => {
    const wrapper = shallowMount(Navbar, {
      stubs: { RouterLink: RouterLinkStub }
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('should not render login buttons if logged in', () => {
    const Parent = {
      data () {
        return {
          isLoggedIn: true
        }
      }
    }
    const wrapper = shallowMount(Navbar, {
      stubs: { RouterLink: RouterLinkStub },
      parentComponent: Parent
    })
    expect(wrapper.find('#login-buttons').exists()).toBeFalsy()
  })

  it('should show login form on login click', () => {
    const wrapper = shallowMount(Navbar, {
      stubs: { RouterLink: RouterLinkStub }
    })

    expect(wrapper.vm.$data.showLogin).toBeFalsy()
    wrapper.find('#login').trigger('click')
    expect(wrapper.vm.$data.showLogin).toBeTruthy()
  })
})
