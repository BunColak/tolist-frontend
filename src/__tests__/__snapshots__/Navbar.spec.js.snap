// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render without problems 1`] = `
<nav class="p-4 bg-indigo-700 text-white">
  <div class="container mx-auto flex justify-between items-center">
    <div class="flex"><a class="font-semibold text-lg">ToList</a>
      <div><a class="text-lg ml-8">Features</a> <a class="text-lg ml-4">Templates</a></div>
    </div>
    <div id="login-buttons"><button id="login" class="mx-4">Login</button> <button id="register" class="mx-4 border-2 px-2 py-1 rounded">Signup</button></div>
  </div>
  <login-stub style="display: none;"></login-stub>
</nav>
`;
